src/Transport.php

```diff
@@ -28,7 +28,10 @@ protected function doSend(SentMessage $message): void
     {
         $token = $this->getToken();
         $email = MessageConverter::toEmail($message->getOriginalMessage());
-        $url = sprintf('https://graph.microsoft.com/v1.0/users/%s/sendMail', $email->getFrom()[0]->getEncodedAddress());
+        $url = sprintf(
+            'https://graph.microsoft.com/v1.0/users/%s/sendMail',
+            $this->config['username'] ? urlencode($this->config['username']) : $email->getFrom()[0]->getEncodedAddress()
```

Comment:
Won't this fail if "username" is not set?

Reply:
Probably